File: backend/model/momoModel.js
const mongoose = require("mongoose");
const Rating = require("./ratingModel");
const User = require("./userModel");

const momoSchema = mongoose.Schema({
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: User,
    required: true,
  },
  momoName: {
    type: String,
    required: true,
  },
  momoImage: {
    type: String,
    required: true,
  },
  momoPrice: {
    type: Number,
    required: true,
  },
  cookType: {
    type: String,
    required: true,
  },
  fillingType: {
    type: String,
    required: true,
  },
  location: {
    type: String,
    required: true,
  },
  shop: {
    type: String,
    required: true,
  },
  overallRating: {
    type: Number,
  },
  reviews: [
    {
      type: mongoose.Schema.Types.ObjectId,
      ref: Rating,
    },
  ],
});

const Momo = mongoose.model("momo", momoSchema);
module.exports = Momo;

File: frontend/momo_rating_app_frontend/lib/core/model/momo/momo_model.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:momo_rating_app_frontend/core/model/momo/review_model.dart';

// part 'momo_model.g.dart'

// @JsonSerializable()
class MoMoApiModel {
  @JsonKey(name: '_id')
  final String? id;
  final String? momoImage;
  final String userId;
  final String momoName;
  final String momoPrice;
  final String cookType;
  final String fillingType;
  final String location;
  final String shop;
  final bool? isSaved;
  final double? overallRating;
  final List<Review>? reviews;

  MoMoApiModel({
    this.id,
    this.momoImage,
    this.isSaved,
    this.overallRating,
    required this.userId,
    required this.momoName,
    required this.momoPrice,
    required this.cookType,
    required this.fillingType,
    required this.location,
    required this.shop,
    this.reviews,
  });

  factory MoMoApiModel.fromJson(Map<String, dynamic> json) {
    var reviewsFromJson = json['reviews'] as List;
    List<Review> reviewList = reviewsFromJson
        .map((reviewJson) => Review.fromJson(reviewJson, json['_id']))
        .toList();

    return MoMoApiModel(
      id: json['_id'],
      isSaved: json['isSaved'],
      momoImage: json['momoImage'],
      userId: json['userId'],
      momoName: json['momoName'],
      momoPrice: json['momoPrice'].toString(),
      cookType: json['cookType'],
      fillingType: json['fillingType'],
      location: json['location'],
      shop: json['shop'],
      overallRating: json['overallRating'].toDouble(),
      reviews: reviewList,
    );
  }
}
